sklearn/utils/tests/test_validation.py

```diff
@@ -324,6 +324,37 @@ def test_check_array_dtype_warning():
     assert_equal(X_checked.format, 'csr')
 
 
+def test_check_array_accept_sparse_type_exception():
+    X = [[1, 2], [3, 4]]
+    X_csr = sp.csr_matrix(X)
+    invalid_type = SVR()
+
+    msg = "A sparse matrix was passed, but dense data is required. " \
+        "Use X.toarray() to convert to a dense numpy array."
+    assert_raise_message(TypeError, msg,
+                         check_array, X_csr, accept_sparse=False)
+    assert_raise_message(TypeError, msg,
+                         check_array, X_csr, accept_sparse=None)
+
+    msg = "The parameter 'accept_sparse={}' was not a correct type."
+    assert_raise_message(ValueError, msg.format([]),
+                         check_array, X_csr, accept_sparse=[])
+    assert_raise_message(ValueError, msg.format(()),
+                         check_array, X_csr, accept_sparse=())
+    assert_raise_message(ValueError, msg.format(invalid_type),
+                         check_array, X_csr, accept_sparse=invalid_type)
+
+    msg = "Can't convert 'SVR' object to str implicitly"
```

Comment:
Just use ``"'SVR' object"`` and it'll work on python2 and python3 and we know it's the right error (probably)